cmake/KokkosPrefix.cmake

```diff
@@ -0,0 +1,74 @@
+#=========================================================================================
+# (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+#
+# This program was produced under U.S. Government contract 89233218CNA000001 for Los
+# Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+# for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+# in the program are reserved by Triad National Security, LLC, and the U.S. Department
+# of Energy/National Nuclear Security Administration. The Government is granted for
+# itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+# license in this material to reproduce, prepare derivative works, distribute copies to
+# the public, perform publicly and display publicly, and to permit others to do so.
+#=========================================================================================
+
+# This file swaps out the specified C++ compiler for nvcc_wrapper. This is done because
+# nvcc_wrapper is not a real compiler, and is just a workaround for the fact that nvcc
+# doesn't like to compile .cpp files as Cuda.
+
+if (NOT Kokkos_ENABLE_CUDA)
+    # TODO: Support installs of Kokkos. This will have to be done by looking up the
+    # Kokkos settings and checking if CUDA is enabled.
+    return()
+endif()
+
+# Cache the environment-provided C++ compiler
+if (NOT DEFINED PARTHENON_HOST_CXX_COMPILER)
+    if (CMAKE_CXX_COMPILER)
+        set(CXX_COMPILER ${CMAKE_CXX_COMPILER})
+    elseif(DEFINED ENV{CXX})
+        set(CXX_COMPILER $ENV{CXX})
+    endif()
+    find_program(PARTHENON_HOST_CXX_COMPILER NAMES ${CXX_COMPILER})
+endif()
+
+# Check if the user specified nvcc_wrapper for the host compiler and issue a warning.
+if (PARTHENON_HOST_CXX_COMPILER MATCHES "nvcc_wrapper")
+    message(
+        WARNING
+        "Parthenon can automatically apply nvcc_wrapper. It is recommended you just \
+        allow nvcc_wrapper to be detected.")
+    unset(PARTHENON_HOST_CXX_COMPILER CACHE)
+    return()
+endif()
```

Comment:
I'm glad to see it's possible to override the automatic selection, as there may be times when this is necessary.